README.md

```diff
@@ -172,35 +173,50 @@ run:
 iex> Nerves.Runtime.revert
 ```
 
-Going back to previous versions of firmware is an important topic for building
-devices that can survive buggy firmware updates without manual intervention.
-Making this work well involves non-Elixir components like bootloaders. This
-feature isn't intended to be bulletproof, but it certainly can get you out of
-bad situations.
-
-One important use case is to be able to remotely update a device and have it
-automatically revert its firmware after a timeout or if it can't reach the
-network. A common requirement is to handle crashes and hangs. The noraml strategy for
-implementing this to have the device allow one boot of new firmware and then to
-mark it "valid" if code gets to a good point (like connect to a server). If
-something goes wrong, then the next reboot reverts back to the original
-firmware. If you're running a scriptable bootloader like U-boot, it's
-best to have the logic implemented there to minimize the code that must work.
-Here's a simple alternative:
-
-1. After upgrading firmware, save that the next boot is the first one.
-1. On the reboot, if this is the first boot, record that the boot happened and
+Running this command manually is useful in development. Production use requires
+more work to protect against faulty upgrades.
+
+### Assisted firmware validation and automatic revert
+
+Some Nerves systems all your application to determine whether newly installed
+firmware is good or not.  For example, a test for "good" could be "can a new
+firmware reconnect to the firmware update server in 5 minutes?" If the answer to
+that question is negative, the device should reboot back to the previously
+running firmware.
+
+Nerves systems that support the feature work as follows:
+
+1. New firmware is installed in the normal manner. The `Nerves.Runtime.KV`
+   variable, `nerves_validated` is set to 0.
+2. The system reboots like normal.
+3. The application starts a five minute reboot timer (this need not be the
+   application, but it is easier to understand)
+4. The application attempts to make a connection to the firmware update server.
+5. On a good connection, the application sets `nerves_validated` to 1 and
+   cancels the reboot timer.
+6. On error, the reboot timer failing, or a hardware watchdog timeout, the
+   system reboots. The bootloader reverts to the previous firmware.
+
+To use this feature, the `nerves_autovalidate` variable must be set to 0. This
+can be done at device provisioning time when other keys (like the serial number)
+are set or a custom `fwup.conf` can be made to force it.
```

Comment:
> are set or a custom `fwup.conf` can be made to force it

Feels like there should be a comma here?
are set or, you could create a custom `fwup.conf` to force setting `nerves_autovalidate`.